### ISSUE TITLE
Feature Request: Make X++ Backgrounds accessible for Plugins

### ISSUE THREAD

**Issue by 6xthFNGR** [xournalpp/xournalpp#6106]
### Feature Description

As requested in feature https://github.com/xournalpp/xournalpp/issues/1174 and by myself in feature https://github.com/xournalpp/xournalpp/issues/6085, there is a clear need for extending PDF background dimensions to enhance annotation space.

Fortunately, MiltonBalaOfficial developed a plugin to extend PDF dimensions: https://github.com/MiltonBalaOfficial/IncreaseAndDecreasePageDimensions. Additionally, Robin-Sch created another plugin, https://github.com/Robin-Sch/Xournalpp-Infinity-Notes, to simulate an infinite canvas. These provide excellent workarounds for enhancing Xournal++ features until this functionality can be integrated into the Xournal++ core.

When using the https://github.com/MiltonBalaOfficial/IncreaseAndDecreasePageDimensions plugin to increase PDF dimensions, an issue arises where empty space is created. This issue is documented in https://github.com/MiltonBalaOfficial/IncreaseAndDecreasePageDimensions/issues/2.
The ultimate solution would be to fill this empty space with Xournal++ backgrounds, but this is currently not possible.

Milton provided another workaround by enhancing the plugin to let users toggle whether the empty space should automatically be filled with the pen color. This is a great short-term solution to provide at least some usability.

While the plugin has matured significantly within just one week, it is now limited by the features Xournal++ exposes to plugins. Therefore, it would be highly beneficial to extend plugin functionality to allow access to Xournal++’s patterned backgrounds. These backgrounds are particularly useful, especially for features like Snapping to Grid.

**Comment by MiltonBalaOfficial** [comment]
Yes, at least make changes so that below the pdf background there remains an additional layer having Xournal++ background (having grids) if possible. grid snapping works on the empty space but grids be very helpful. Please consider the issue.
![image](https://github.com/user-attachments/assets/c64600dd-9bad-4cc1-b2ea-4da0a3d9f258)
